strawberryfields/backends/bosonicbackend/bosoniccircuit.py

```diff
@@ -337,15 +467,24 @@ def fidelity_coherent(self, alpha, modes=None):
         return fidelity
 
     def fidelity_vacuum(self, modes=None):
-        """fidelity of the current state with the vacuum state"""
+        r"""Returns the fidelity to the vacuum.
+
+        Args:
+            modes (list or None): modes to use for fidelity calculation
+        """
         if modes is None:
-            modes = list(range(self.nlen))
+            modes = self.get_modes()
         alpha = np.zeros(len(modes))
         fidelity = self.fidelity_coherent(alpha, modes=modes)
         return fidelity
 
     def parity_val(self, modes=None):
-        """Expectation value of the parity operator"""
+        r"""Returns the expectation value of the parity operator, which is the
+        value of the Wigner function at the origin.
+
+        Args:
+            modes (list or None): modes to use for parity calculation
+        """
         if modes is None:
```

Comment:
wondering if we need an `hbar` somewhere in here. The value of parity is hbar independent, while the covariance matrices themselves are. Probably there has to be an `hbar`-depedent scaling of the value of the Wigner function to get the parity.

Reply:
You are correct! I had changed it in the calculation of parity in states.py, but had forgotten it here. Will add now.